editors/code/src/main.ts

```diff
@@ -109,3 +99,130 @@ export async function deactivate() {
     await ctx?.client?.stop();
     ctx = undefined;
 }
+
+async function bootstrap(config: Config, state: PersistentState): Promise<string> {
+    await bootstrapExtension(config, state);
+    const path = await bootstrapServer(config, state);
+    return path;
+}
+
+async function bootstrapExtension(config: Config, state: PersistentState): Promise<void> {
+    if (config.channel === "stable") {
+        if (config.extensionReleaseTag === NIGHTLY_TAG) {
+            vscode.window.showWarningMessage(`You are running a nightly version of rust-analyzer extension.
+To switch to stable, uninstall the extension and re-install it from the marketplace`);
```

Comment:
I'd suggest testing that this does work. I.e. uninstalling and installing does switch us to `stable`.
(see my previous review summary comment https://github.com/rust-analyzer/rust-analyzer/pull/3629#pullrequestreview-376484185)

Reply:
yup, verified manually that it works.